ckan_cloud_operator/providers/cluster/azure/manager.py

```diff
@@ -0,0 +1,130 @@
+#### standard provider code ####
+
+import subprocess
+
+# import the correct PROVIDER_SUBMODULE and PROVIDER_ID constants for your provider
+from .constants import PROVIDER_ID
+from ..constants import PROVIDER_SUBMODULE
+
+# define common provider functions based on the constants
+from ckan_cloud_operator.providers import manager as providers_manager
+def _get_resource_name(suffix=None): return providers_manager.get_resource_name(PROVIDER_SUBMODULE, PROVIDER_ID, suffix=suffix)
+def _get_resource_labels(for_deployment=False): return providers_manager.get_resource_labels(PROVIDER_SUBMODULE, PROVIDER_ID, for_deployment=for_deployment)
+def _get_resource_annotations(suffix=None): return providers_manager.get_resource_annotations(PROVIDER_SUBMODULE, PROVIDER_ID, suffix=suffix)
+def _set_provider(): providers_manager.set_provider(PROVIDER_SUBMODULE, PROVIDER_ID)
+def _config_set(key=None, value=None, values=None, namespace=None, is_secret=False, suffix=None): providers_manager.config_set(PROVIDER_SUBMODULE, PROVIDER_ID, key=key, value=value, values=values, namespace=namespace, is_secret=is_secret, suffix=suffix)
+def _config_get(key=None, default=None, required=False, namespace=None, is_secret=False, suffix=None): return providers_manager.config_get(PROVIDER_SUBMODULE, PROVIDER_ID, key=key, default=default, required=required, namespace=namespace, is_secret=is_secret, suffix=suffix)
+def _config_interactive_set(default_values, namespace=None, is_secret=False, suffix=None, from_file=False): providers_manager.config_interactive_set(PROVIDER_SUBMODULE, PROVIDER_ID, default_values, namespace, is_secret, suffix, from_file)
+
+################################
+# custom provider code starts here
+#
+
+import binascii
+import json
+import os
+import yaml
+
+from ckan_cloud_operator import logs
+from ckan_cloud_operator import kubectl
+
+
+def initialize(interactive=False):
+    _set_provider()
+    if interactive:
+        print('\nEnter the Resource Group name\n')
+        _config_interactive_set({'azure-rg': None}, is_secret=False)
+        print('\nEnter the location of the Kubernets cluster is hosted on [westus2]\n')
+        _config_interactive_set({'azure-default-location': None}, is_secret=False)
+        print('\nEnter the name of your cluster\n')
+        _config_interactive_set({'azure-cluster-name': None}, is_secret=False)
```

Comment:
What if not interactive? Will it just skip and do nothing? Should not we at least say smth like "please set --interactive flag" or similar?

Reply:
providers_manager.config_interactive_set has an 'interactive' parameter, better use it instead - that is, remove the if and pass interactive to every call of `_config_interactive_set`